plugins/vectors/IVectorDB.py

```diff
@@ -0,0 +1,31 @@
+from abc import ABC, abstractmethod
+
+class VectorDB(ABC):
+    @abstractmethod
+    def create_table(self, name: str, vectorizer: str) -> bool:
+        pass
+
+    @abstractmethod
+    def delete_table(self, name: str) -> bool:
+        pass
+
+    @abstractmethod
+    def has_table(self, name: str) -> bool:
+        pass
+
+    @abstractmethod
+    def query_table(self, query: str, table_name: str, n: int) -> list:
```

Comment:
This seems fairly generic, and looks like create_table and has_table is only done in init. 
Should we make this a lot more simpler, where the VectorDB class is responsible for handling the init (or just a single init?).
Query table is also fairly generic, which I think then your query is also specific to the vectordb. Might be better if we make this specific to the baby and each implementation can translate it to the query it targets